### PR TITLE
Remove GH action warnings

### LINKED ISSUE #229: Upgrade to Environment Files
We get the following warning on our actions ([example](https://github.com/PolicyEngine/policyengine-api/actions/runs/4351629443)):
>The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands

### PR DESCRIPTION
**Pull request by anth-volk**
Fixes #229 

Solution: Upgraded [EndBug/add-and-commit](https://www.github.com/EndBug/add-and-commit), used within push.yml, to version 9, as described in [this](https://github.com/EndBug/add-and-commit/issues/448) issue solution within the EndBug/add-and-commit repo. 

Version 9 does introduce two breaking changes (described [here](https://github.com/EndBug/add-and-commit/releases)): the action fails if `git commit` fails, and the action will not attempt to recreate tags when failing to push them. However, it was not possible to test for these two cases.

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Thanks @anth-volk! Could you please:
* Replace `Fix for: #229` with `Fixes #229` in the PR description; this specific syntax will link the PR to the issue so merging it will close the issue
* Populate `changelog_entry.yaml` per other recently merged PRs to pass the version check CI test.

**Comment by anth-volk** [comment]
Ah, my apologies, I should've reviewed the PR requirements before requesting

**Comment by anth-volk** [comment]
I'm sorry, I'm not understanding the changelog system. I altered `changelog_entry.yaml` as instructed, adding the details of the relevant changes, but the version check failed, indicating that I hadn't also updated `CHANGELOG.md`. Is this not automatically generated? I was under the impression that `changelog_entry.yaml` was automatically pushed to the top of the file, with the relevant change type updating the relevant portion of the semantic version number. Should `CHANGELOG.md` (and `changelog.yaml`, for that matter) also be manually updated?

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: patch
+  changes:
+    fixed:
+    - EndBug/add-and-commit updated to remove git commit warnings
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
      - EndBug/add-and-commit updated to remove git commit warnings.
```

**Review by MaxGhenis — COMMENTED** [review]
You populated the changelog entry correctly, just this minor change and `git pull upstream master` will do it. The action fails if the branch is behind upstream/master.

File: changelog_entry.yaml (hunk 1 above)
Comment (anth-volk) [reply changelog_entry.yaml:4]: Ah, that makes sense, thank you, and sorry for all of the hassle throughout.

**Comment by anth-volk** [comment]
This should now be properly synced with upstream/main and integrate the typographical changes suggested; however, I'm still receiving the same issue when running checks. I'm going to try to dive into this and figure out what it might be.

**Comment by anth-volk** [comment]
It appears that, when forking, Git does not also copy upstream's tags, so after fetching tags from upsteram master, I should now be able to pass the has-functional-changes GitHub workflow

**Comment by nikhilwoodruff** [comment]
Sorry @anth-volk - could you re-file from a branch? I just gave you write access- happy to help if unclear

**Comment by nikhilwoodruff** [comment]
Actually... this is mergeable- I've just checked the file change. Thanks Anthony!

**Comment by anth-volk** [comment]
Great, thanks! Glad to have finally gotten this one in haha. Thanks for all the support, @nikhilwoodruff and @MaxGhenis.

**Pull request merged**